plugins/logistics.cpp

```diff
@@ -54,6 +54,7 @@ enum StockpileConfigValues {
     STOCKPILE_CONFIG_DUMP = 3,
     STOCKPILE_CONFIG_TRAIN = 4,
     STOCKPILE_CONFIG_MELT_MASTERWORKS = 5,
+    STOCKPILE_CONFIG_FORBID = 6,
```

Comment:
existing savegames will have this variable set to -1, which this PR does not seem to handle gracefully. if the new config were two bools, they'd default to false, but an int needs to be initialized explicitly.

Reply:
Thanks - I'll take a closer look, in my testing I saw the negative 1 get cleaned up (and I thought that was due to the initialization in ensure_stockpile config - but thinking about it now I suspect that may not get called until after stockpiles.lua is trying to use the value with an invalid index...

Reply:
Still reviewing/resolving this one as well.

Reply:
I did this in `validate_stockpile_configs` as that seemed to address it but please let me know if you see a loophole or a better place to shore it up. The original behavior was dodging any negative effects because it was always looking for values of 1 or 2 without care for other values; but I agree that was not great to leave that way.